vignettes/cpp11_faq.Rmd

```diff
@@ -0,0 +1,184 @@
+---
+title: "cpp11 FAQ"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{cpp11 FAQ}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+Intended to answer common questions about the usage of cpp11 objects
+
+## FAQ
+
+* How do I modify a vector in place?
+
+  * Using `writable::` will always make a copy, but it has a move constructor, so you can use `cpp11::writable::integers(std::move(x))` and it won't make a copy of the data.
+
+* How do I add elements to a named list?
+  
+  ```
+  cpp11::cpp_source(code = '#include <cpp11.hpp>
+  using namespace cpp11::literals;
+  [[cpp11::register]]
+  cpp11::list fn() {
+    cpp11::writable::list x;
+    x.push_back({"foo"_nm = 1});
+    return x;
+  }')
+  fn()
+  #> $foo
+  #> [1] 1
+  ```
+
+* How do I create a new empty environment?
+
+  ```
+  auto new_env = cpp11::package("base")["new.env"];
+  cpp11::environment my_env(new_env());
+  ```
+
+* How do I assign and retrieve values in an environment? What happens if I try to get a value that doesn't exist?
+
+  ```
+  my_env["foo"] = 1;
+  my_env["foo"] == 1;
+
+  my_env["bar"] == R_UnboundValue;
+  ```
+
+* How can I create a `cpp11:raws` from a `std::string`?
+
+  * Push back the contents individually
+  ```
+  cpp11::cpp_source(code = '#include <cpp11.hpp>
+    [[cpp11::register]]
+    cpp11::raws fn() {
+      std::string x("hi");
+      cpp11::writable::raws out;
+      for (auto c : x) {
+        out.push_back(c);
+      }
+      return out;
+    }')
+  fn()
+  #> [1] 68 69
+  ```
+  
+* What are the underlying types of cpp11 objects?
+
+    vector   | element
+    ---      | ---
+    integers |	int
+    doubles	 | double
+    logical  | r_bool
+    strings	 | r_string
+    raws	   | uint8_t
+    list	   | SEXP
+
+* How do I create a new empty list?
+
+    `cpp11::writable::list x;`
+
+* How do I retrieve (named) elements from a named vector/list?
+
+    `x["foo"]`
+
+* How can I tell whether a vector is named?
+
+    `x.names() == R_NilValue;`
```

Comment:
I think we should also open an issue to add a `named()` function, for the same reasons as #182

Reply:
That's a great idea! I'll open an issue for this.